biblib/views/user_view.py

```diff
@@ -85,13 +85,8 @@ def get_libraries(cls, service_uid, absolute_uid, start=0, rows=None, sort_col="
                 .order_by(getattr(getattr(Library, sort_col), sort_order)())\
                 .all()
             
-            libraries_response = {'libraries_count': len(user_libraries)}
-            
-            if rows: rows=start+rows
-            
-            my_libraries = []
-            shared_with_me = []
-            for permission, library in user_libraries[start:rows]:
+            libraries = []
+            for permission, library in user_libraries:
```

Comment:
For someone like Mike who has over 1000 libraries, just thinking of efficiency: 
* Is it possible to do the filtering of type in the query on line 82? Then it would be easy to get the subset using `start` and `rows`, and we would have the total count too.

Reply:
Yeah. I was also concerned about efficiency. I didn't want to change too much of the code, but let me take another stab at it.

Reply:
I changed the query.